src/PolyType.Roslyn/Model/TypeExtensionModel.cs

```diff
@@ -0,0 +1,20 @@
+﻿using Microsoft.CodeAnalysis;
+using System.Collections.Immutable;
+
+namespace PolyType.Roslyn;
+
+/// <summary>
+/// Captures the data collected from all the TypeShapeExtensionAttribute attributes that target a particular type.
+/// </summary>
+public sealed record TypeExtensionModel
```

Comment:
Likewise, this would be more of a `TypeShape.SourceGenerator` concept as opposed to a `TypeShape.Roslyn` concept. What I've been doing for things like surrogates is define extensions on the `TypeShape.Roslyn` model types in the source generator alongside the incremental model:

https://github.com/eiriktsarpalis/PolyType/blob/24ea2d1adc44c79af2a0d3e17f0d24ac16e6854e/src/PolyType.SourceGenerator/Model/SurrogateShapeModel.cs#L12-L16